stable/admin/files/readinessprobe

```diff
@@ -0,0 +1,37 @@
+#!/bin/sh
+
+nuocmd_fallback() {
+    # run nuocmd subcommand and suppress parse errors due to the command
+    # containing unsupported arguments or the subcommand not being supported at
+    # all by the available version of nuocmd; if the command succeeds or fails
+    # with a non-parse error, emit command output and exit
+
+    # capture stdout and stderr and get exit code
+    out="$(nuocmd "$@" 2>&1)"
+    ret=$?
+
+    # nuocmd returns exit code 2 if there is a parse error due to unrecognized
+    # subcommand or arguments
+    if [ $ret != 2 ]; then
+        # the command succeeded or failed with a non-parse error; emit command
+        # output and exit
+        echo "$out"
+        exit $ret
+    fi
+}
+
+# 'nuocmd check server' is introduced in versions >4.1.1; this checks that the
+# --api-server satisfies the following conditions:
+#   1. It is ACTIVE and able to service REST requests.
+#   2. It is able to ping its own advertised address, i.e. altAddr in
+#      nuoadmin.conf, which is derived from the --alt-address argument of
+#      'nuodocker start admin' and the NUODB_ALT_ADDRESS environment variable
+#      for the nuoadmin entrypoint script.
+#   3. It has the same commit index as the current Raft leader, which means
+#      that its Raft state is up-to-date.
+nuocmd_fallback check server --check-active --check-connected --check-converged
+
+# 'nuocmd check server' is unsupported; use 'nuocmd check servers' (plural)
+# with no arguments, which simply checks that the --api-server is ACTIVE and
+# able to service REST requests
+nuocmd check servers
```

Comment:
I just wanna point out that in pre-4.1.1 NuoDB this is a change in behavior. This probe is laxer than it used to be.

Reply:
Yes, this leaves out `--check-connected` and `--check-leader`, which actually check all processes in the domain, not just the process being probed, as I mentioned in the PR summary:

> Previously the readiness probe for Admin Pods checked for full-server connectivity and that all Admin processes agree on a Raft leader. Since these are not properties of a particular Admin process, but of the entire domain, they were not appropriate checks for a readiness probe (e.g. admin-1 could show as not ready because admin-2 is having connectivity issues).

We'll see if this actually breaks anything.